docs/src/modules/java/pages/agents.adoc

```diff
@@ -701,6 +701,47 @@ include::example$multi-agent/src/test/java/demo/multiagent/application/AgentTeam
 <5> For more complex scenarios, define a response that is only returned if the user message matches a specific condition. This is useful for testing different logic paths within your agent.
 <6> Call the components with the `componentClient`.
 
+=== Mocked model in a deployed service
+
+In some scenarios it can be useful to run the service deployed but without interacting with an actual agent. For example,
+a load test that exercises the service with heavy load to verify scalability could quickly consume a large number of tokens
+when the exact answer from the model is not very important, one or a few different predefined responses and responding with
+a slight delay to simulate model processing time could be good enough.
+
+It is possible to implement a custom model provider using `akka.javasdk.agent.ModelProvider.Custom`, such a mock provider
+however, side steps quite a bit of the infrastructure involved in agent interactions, a more realistic mock model can be implemented
+by building a separate Akka service with a single xref:http-endpoints.adoc[HTTP endpoint] mimicking the model endpoint and
+configuring the deployed agentic service to use that.
+
+Here is an example endpoint returning a static response over the OpenAI protocol:
+
+[source,java,indent=0]
+.{sample-base-url}/doc-snippets/src/main/java/com/example/api/MockOpenAI.java[tag=class]
+----
+include::example$doc-snippets/src/main/java/com/example/api/MockOpenAI.java[tag=class]
+----
+
+For more elaborate scenarios, the mock model endpoint may have to parse the request to decide which hardcoded answer out of a few
+or to create a reply in a more dynamic fashion.
+
+Deploying this service as `mock-openai` allows a service containing agents using it with a config like this:
```

Comment:
mention something about that they must be deployed in the same project (and link to what a project is)